backend/src/test/java/com/github/alaajouri/backend/controller/MongoUserControllerTest.java

```diff
@@ -0,0 +1,197 @@
+    }
+
+
+    @Test
+    @DirtiesContext
+    void createUserWithoutUsername_ThenBadRequest400() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/user")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {
+                                    "username": "",
+                                    "password": "test"
+                                }
+                                """)
+                        .with(csrf()))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    @DirtiesContext
+    void createUserWithoutPassword_ThenBadRequest400() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/user")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {
+                                    "username": "test",
+                                    "password": ""
+                                }
+                                """)
+                        .with(csrf()))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    @DirtiesContext
+    void createUser_UserAlreadyExists() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/user")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {
+                                    "username": "test",
+                                    "password": "test"
+                                }
+                                """)
+                        .with(csrf()))
+                .andExpect(status().isOk());
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/user")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {
+                                    "username": "test",
+                                    "password": "test"
+                                }
+                                """)
+                        .with(csrf()))
+                .andExpect(status().isConflict());
+    }
+
+    @Test
+    @DirtiesContext
+    @WithMockUser(username = "user", password = "password")
+    void loginUserWithValidUsernameAndPassword_Then200() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/user/login")
+                .with(csrf())
+                .contentType(MediaType.APPLICATION_JSON)
+                .content("""
+                        {
+                            "username": "user",
+                            "password": "password"
+                        }
+                        """));
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/user")
+                        .with(csrf()))
+                .andExpect(status().isOk());
```

Comment:
You should only make one request per test. Here you are only testing the get api/user endpoint. 
For this reason SonarCloud is telling you that login is not tested. You need to delete the second mockMvc request and add an assertion to the first request. Assert, that it returns a JSON containing the user.